src/xmipp/libraries/data/image_operate.cpp

```diff
@@ -531,7 +531,8 @@ void ProgOperate::readParams()
             {
                 if (mdInSize != md2.size())
                     REPORT_ERROR(ERR_MD, "Both metadatas operands should be of same size.");
-                md2Iterator.init(md2);
+                md2IdIterator = std::unique_ptr<MetaDataVec::id_iterator>(new MetaDataVec::id_iterator(md2.ids().begin()));
+                md2.ids().begin();
```

Comment:
What does this do?

Reply:
Initializes iterator to `begin` of `MetaData`. Code simplified using `make_unique`.